src/Dialect/ONNX/ONNXOps/Additional/LayoutTransform.cpp

```diff
@@ -42,10 +38,19 @@ LogicalResult ONNXLayoutTransformOp::inferShapes(
 // Verifier
 //===----------------------------------------------------------------------===//
 LogicalResult ONNXLayoutTransformOp::verify() {
-  if (hasShapeAndRank(getData()) && hasShapeAndRank(getOutput())) {
-    if (getShape(getData().getType()) != getShape(getOutput().getType()))
-      return emitOpError("Input and output tensors must have the same shape");
-  }
+  if (auto dataType = getData().getType().dyn_cast<RankedTensorType>())
+    if (auto outputType = getOutput().getType().dyn_cast<RankedTensorType>())
+      for (int64_t i = 0; i < dataType.getRank(); ++i)
+        // Check if there is an unknown dimension in the dataShape and
+        // outputShape. If there is an unknown dimension, we will return true.
+        // If we know the dimension of dataShape and outputShape they should be
+        // equal, if not then we return false.
+        if (dataType.getShape()[i] == ShapedType::kDynamic ||
+            outputType.getShape()[i] == ShapedType::kDynamic)
+          return success();
+        else if (dataType.getShape()[i] != outputType.getShape()[i])
```

Comment:
@hamptonm1 this causes a compiler warning
```
LayoutTransform.cpp:51:9: warning: add explicit braces to avoid dangling else [-Wdangling-else]
        else if (dataType.getShape()[i] != outputType.getShape()[i])
```

Reply:
here are LLVM's guidelines for when to (not) omit curly braces:
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements